### PR TITLE
Apply Wolfi Cloud Instance Fixes

### PR DESCRIPTION
**Pull request by willdollman**
This PR contains all the fixes and tweaks I've made during the process of getting the wolfi-test cloud instance working.

There's a lot going on in this PR - it probably should've been broken up, but CI issues made that difficult to do. Future PRs will be smaller :) 

Packages:
* Build comby from source - the published binary isn't compatible with Arch/Wolfi's libpcre library path
* Build cadvisor from source - may not be necessary, but done as a debugging step and gives us more arch flexibility
* Package s3proxy
* Update p4-fusion dependency SHA
* Bump all packages to trigger a rebuild with latest version of melange. The old version didn't seem to pick up on dylibs properly
* Fix opentelemetry-collector binary path

Base images:
* Fix cadvisor base image
* Fix redis UID+GIDs
* Add missing p4cli package to repo-updater image
* Remove nodejs from server image (no longer required by code-intel)

Other fixes:
* Update build-wolfi.sh scripts to match the current Bazelified build.sh scripts
* Fix `server` image builds
* Fix zoekt-webserver CMD
* Fix postgres locale + default config + add a patch for `unix_socket_directories`
* Fix `symbols` go-build-wolfi scripts

## Things to do before merging

- [ ] Validate the new postgres patch-conf.sh script with delivery
- [x] Remove all DO NOT MERGE comments

## Test plan

- [x] Manual testing on wolfi-test cloud instance
- [x] Manual testing on scaletesting instance

### CODE REVIEW

File: wolfi-packages/cadvisor.yaml [hunk 1]
```diff
@@ -23,14 +30,45 @@ environment:
...
+      - thin-provisioning-tools
 
 pipeline:
+  # Fetch and build libpfm
   - uses: fetch
     with:
-      uri: https://github.com/google/cadvisor/releases/download/v${{package.version}}/cadvisor-v${{package.version}}-linux-amd64
-      expected-sha256: caf4491298e0702f9d0c6a1d1949767f5c6400f77e12cd3524d6d3fcc66abc2a
-      extract: false
+      uri: https://sourceforge.net/projects/perfmon2/files/libpfm4/libpfm-4.11.0.tar.gz
```
Comment (jhchabran) [inline wolfi-packages/cadvisor.yaml:47]: Small thing, is there a more reliable place to download this from? Sourceforge is notorious for injecting crap in tarballs :/

File: enterprise/dev/ci/internal/ci/pipeline.go [hunk 2]
```diff
@@ -127,42 +127,14 @@ func GeneratePipeline(c Config) (*bk.Pipeline, error) {
...
-				"redis-store",
-				"redis_exporter",
-				"search-indexer",
-				"sg",
-				"syntax-highlighter",
+				"cadvisor",
 			}, c.Version,
-				c.candidateImageTag(),
+				// c.candidateImageTag(),
+				"latest", // TODO: Do not merge; used to simplify dev
```
Comment (jhchabran) [inline enterprise/dev/ci/internal/ci/pipeline.go:137]: @willdollman is the TODO still being here intentional?

**Review by jhchabran — CHANGES_REQUESTED** [review]

File: wolfi-packages/cadvisor.yaml (hunk 1 above)
Comment (willdollman) [reply wolfi-packages/cadvisor.yaml:47]: There's a [github mirror](https://github.com/wcohen/libpfm4) but it's missing release tags for recent releases. We could pin a commit but that's quite icky.

The official Google cadvisor image build [uses a sourceforge tarball](https://github.com/google/cadvisor/blob/master/deploy/Dockerfile#L10) and the hash means we'd notice if it changes. Do you think that's good enough?

File: enterprise/dev/ci/internal/ci/pipeline.go (hunk 2 above)
Comment (willdollman) [reply enterprise/dev/ci/internal/ci/pipeline.go:137]: 👍 I'm currently using this to speed up CI by only rebuilding specific images. Before merging I need to revert this change so it includes all Wolfi images again.

File: wolfi-packages/cadvisor.yaml (hunk 1 above)
Comment (jhchabran) [reply wolfi-packages/cadvisor.yaml:47]: Ok, you're right as long as we got the hash we're good :)

File: enterprise/dev/ci/internal/ci/pipeline.go (hunk 2 above)
Comment (willdollman) [reply enterprise/dev/ci/internal/ci/pipeline.go]: This has now been removed.

**Pull request merged**